library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -268,8 +267,5 @@ class ProtocolClient(
         )
     }
 
-    private fun <Input : Any, Output : Any> urlFromMethodSpec(methodSpec: MethodSpec<Input, Output>): URL {
-        val host = config.baseUri.resolve("/${methodSpec.path}")
-        return host.toURL()
-    }
+    private fun urlFromMethodSpec(methodSpec: MethodSpec<*, *>) = config.baseUri.resolve(methodSpec.path).toURL()
```

Comment:
This does the right thing in my testing:

* baseUri=`https://connectrpc.com`, path=`connectrpc.eliza.v1.ElizaService/Say`, url=`https://connectrpc.com/connectrpc.eliza.v1.ElizaService/Say`
* baseUri=`https://connectrpc.com/`, path=`connectrpc.eliza.v1.ElizaService/Say`, url=`https://connectrpc.com/connectrpc.eliza.v1.ElizaService/Say`
* baseUri=`https://connectrpc.com/api`, path=`connectrpc.eliza.v1.ElizaService/Say`, url=`https://connectrpc.com/connectrpc.eliza.v1.ElizaService/Say`
* baseUri=`https://connectrpc.com/api/`, path=`connectrpc.eliza.v1.ElizaService/Say`, url=`https://connectrpc.com/api/connectrpc.eliza.v1.ElizaService/Say`

Reply:
Updated to preserve relative path when building URLs, so the 3rd and 4th example above behave the same.